app/Http/Controllers/ScoresController.php

```diff
@@ -117,14 +119,12 @@ public function userRankLookup()
             }
         }
 
-        $score = ScoreBest
-            ::getClass($params['rulesetId'])
-            ::where([
-                'beatmap_id' => $params['beatmapId'],
-                'hidden' => false,
-                'score' => $params['score'],
-            ])->firstOrFail();
+        $rank = UserRank::getRank(ScoreSearchParams::fromArray([
+            'beatmap_ids' => [$params['beatmapId']],
+            'ruleset_id' => $params['rulesetId'],
+            'before_total_score' => $params['score'],
+        ]));
 
-        return response()->json($score->userRank(['cached' => false]) - 1);
+        return response()->json(($rank ?? 1) - 1);
```

Comment:
actually, why does this return 1 less than the position? 🤔

Reply:
because that's what the original api returns